demo/Render.cpp

```diff
@@ -0,0 +1,774 @@
+/*
+* Copyright (C) 2017 by Godlike
+* This code is licensed under the MIT license (MIT)
+* (http://opensource.org/licenses/MIT)
+*/
+#include <pegasus/Math.hpp>
+#include <demo/Render.hpp>
+
+#include <glbinding/Binding.h>
+
+using namespace pegasus;
+using namespace render;
+
+void mesh::Allocate(Mesh& mesh)
+{
+    //Generate VBO and EBO
+    glGenVertexArrays(1, &mesh.bufferData.vertexArrayObject);
+    glGenBuffers(1, &mesh.bufferData.vertexBufferObject);
+    glGenBuffers(1, &mesh.bufferData.elementBufferObject);
+
+    //Initialize VBO and EBO data
+    glBindVertexArray(mesh.bufferData.vertexArrayObject);
+
+    glBindBuffer(GL_ARRAY_BUFFER, mesh.bufferData.vertexBufferObject);
+    glBufferData(GL_ARRAY_BUFFER, sizeof(GLdouble) * mesh.vertices.size(), &mesh.vertices.front(), GL_STATIC_DRAW);
+
+    glBindBuffer(GL_ELEMENT_ARRAY_BUFFER, mesh.bufferData.elementBufferObject);
+    glBufferData(GL_ELEMENT_ARRAY_BUFFER, sizeof(GLuint) * mesh.indices.size(), &mesh.indices.front(), GL_STATIC_DRAW);
+
+    //Initialize VBO arguments
+    glEnableVertexAttribArray(0);
+    glVertexAttribPointer(0, 3, GL_DOUBLE, GL_FALSE, sizeof(GLdouble) * 6,
+        reinterpret_cast<void*>(sizeof(GLdouble) * 0));
+
+    glEnableVertexAttribArray(1);
+    glVertexAttribPointer(1, 3, GL_DOUBLE, GL_FALSE, sizeof(GLdouble) * 6,
+        reinterpret_cast<void*>(sizeof(GLdouble) * 3));
+
+    glBindVertexArray(0);
+}
+
+void mesh::Deallocate(Mesh& mesh)
+{
+    glDeleteVertexArrays(1, &mesh.bufferData.vertexArrayObject);
+    glDeleteBuffers(1, &mesh.bufferData.vertexBufferObject);
+    glDeleteBuffers(1, &mesh.bufferData.elementBufferObject);
+}
+
+mesh::Mesh mesh::Create(std::vector<GLdouble>&& vertices, std::vector<GLuint>&& indices)
+{
+    Mesh mesh{ std::move(vertices), std::move(indices) };
+    Allocate(mesh);
+    return mesh;
+}
+
+mesh::Mesh mesh::CreateLineSegment(glm::vec3 start, glm::vec3 end)
+{
+    Mesh mesh;
+
+    mesh.vertices = {{
+        start.x, start.y, start.z, 0, 0, 0,
+        end.x, end.y, end.z, 0, 0, 0,
+        end.x, end.y, end.z, 0, 0, 0,
+    }};
+    mesh.indices = {{ 0, 1, 2 }};
+    Allocate(mesh);
+
+    return mesh;
+}
+
+mesh::Mesh mesh::CreatePlane(glm::dvec3 normal, double length)
+{
+    Mesh mesh;
+
+    glm::dvec3 const i = math::CalculateOrthogonalVector(normal) * (length / 2.0);
+    glm::dvec3 const j = glm::normalize(glm::cross(i, normal)) * (length / 2.0);
+
+    mesh.vertices = {{
```

Comment:
`operator=` takes an actual initializer list, no need for two curly braces here. This applies to other places in this file.